documentation/AuthoringSolutionsTutorial.md

```diff
@@ -0,0 +1,342 @@
+
+```json
+        "lifecycleManager": {
+            "start": [
+                "setSettings",
+                {
+                    "type": "gpii.launch.exec",
+                    "command": "\"${{registry}.HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\JAWS15.exe\\}\""
+                }
+            ],
+            "stop": [
+                {
+                    "type": "gpii.launch.exec",
+                    "command": "${{environment}.SystemRoot}\\System32\\taskkill.exe /f /im jfw.exe"
+                },
+                "restoreSettings"
+            ]
+        }
+```
+
+The lifecycle manager accepts 2 arrays keyed by `start` and `stop`, which are invoked when the user logs in and out respectively.  Each array will run it's operation in the order specified in the array, as you may need operations to occur in a very specific order for your solution.
+
+In order to start JAWS, we first want to `setSettings`, so that the users needs and preferences are transformed and stored in the INI file JAWS reads from. After that we use the `gpii.launch.exec` plugin along with a `command` option that will run that executable on the system. Let's take a look at the string specifying the path of the executable.
+
+```json
+"command": "\"${{registry}.HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\JAWS15.exe\\}\""
+```
+
+Here we see another interesting example of string parameters that are available to use.  It so happens that with many windows applications, the actual path to the executable is stored in a stable registry key. This takes care of situations where users may have installed JAWS in a different directory. You can also find the path to it in the registry key at `HKEY_LOCAL_MACHINE -> SOFTWARE -> Microsoft -> Windows -> CurrentVersion -> App Paths -> JAWS15.exe`. In the above the `{registry}` inside the larger `${ }` parameter syntax says that the following will be a registry path. The registry path items are seperated by backslashes, which again need to be escaped in the JSON. Also, the entire string in this case is separated by quotes to take care of spaces in the path. TODO: Is that really why we have to put quotes around it??
+
+When we logout, again we execute a command, in this case called taskkill.exe which is a builtin windows utility to kill tasks and processes. We are instructing it to stop all JAWS processes.  To find taskkill.exe, we lookup the path using another environment variable on Windows called `SystemRoot`, similar to our `APPDATA` environment variable lookup from previously.  Once we've stopped JAWS, we restore the original settings that were on the system.
```

Comment:
"we logout" -> "we log out"
"builtin windows utility" -> "built-in Windows utility"